sql/pgwire/v3.go

```diff
@@ -157,6 +160,25 @@ func (c *v3Conn) parseOptions(data []byte) error {
 }
 
 func (c *v3Conn) serve(authenticationHook func(string, bool) error) error {
+	var tickChan <-chan time.Time
+	{
+		ticker := time.NewTicker(storage.DefaultHeartbeatInterval / 2)
+		tickChan = ticker.C
+		defer ticker.Stop()
+	}
+	go func() {
```

Comment:
Hmm, this extra goroutine makes me think that maybe we should be completely event-driven - as long as `serve` is active, the txn does not time out. When the conn goes down, we call "Unping" (so basically "hey this thing is abandoned right now").